### PR TITLE
Erc20 wrapper

### PR DESCRIPTION
**Pull request by davidbrai**
ERC20Wrapper: an ERC20 wrapper contract for an underlying ERC20 token
ETHWrapper: an ERC20 wrapper contract with ETH as the underlying token

As specified in: https://github.com/yieldprotocol/mentorship2022/issues/3

### CODE REVIEW

File: src/assignment-3/ERC20Wrapper.sol [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import {IERC20} from "yield-utils-v2/token/IERC20.sol";
+
+/// @title ERC20 wrapping an underlying ERC20 token
+/// @author davidbrai
+/// @notice Users can send a pre-specified ERC20 token and received wrapped tokens in return.
+///     The wrapped tokens can be burned in order to withdraw the deposited tokens.
+contract ERC20Wrapper is ERC20Permit {
+
+    /// @notice The wrapped ERC20 token
+    IERC20 public token;
```
Comment (alcueca) [inline src/assignment-3/ERC20Wrapper.sol]: Look about using `immutable` here. It's one of the easiest gas optimizations that can be made, and at the same time very effective.

**Review by alcueca — APPROVED** [review]
Gorgeous, I've got zero complaints. Well done.

File: src/assignment-3/ERC20Wrapper.sol (hunk 1 above)
Comment (davidbrai) [reply src/assignment-3/ERC20Wrapper.sol]: Thanks!

**Comment by davidbrai** [comment]
Arbitrum-Rinkeby deployment:
ERC20Wrapper: https://testnet.arbiscan.io/address/0x7b3831828dbd7c03164ba33bcc1dd9009cba754f
FunToken: https://testnet.arbiscan.io/address/0x2129363821b37000d2c4e986d7ff438397866342
ETHWrapper: https://testnet.arbiscan.io/address/0xe63ccea54d1c53d0ab9444df26c69045ff64ad1d

**Pull request merged**